### ISSUE TITLE
Add unit tests to Notification model

### ISSUE THREAD

**Issue by omarelkashef** [uchicago-cs/chigame#273]
We need to add unit tests and factories for the newly constructed Notification models. Later, we want to test its integration with other models such as FriendInvitation.

**Comment by omarelkashef** [comment]
While working on these test, we realized that the get_by_actor() method in NotificationQuerySet does not handle exceptions that arise from the get method. It should catch and report exceptions when no instances are found and when multiple are found. Once this is implemented, we will need to modify the corresponding tests. Currently, the test only makes sure a generic exception is raised. 
Also, we could not add tests for filter_by_type because that would require us to add factories for other models that can be actors. For now, we are only using FriendInvitation as actor.

**Comment by omarelkashef** [comment]
Closing Statement: We created a number of unit tests for the newly constructed Notification model. The tests target all the methods that could be used to interact with the model. However, we are only testing notification coming from FriendInvitation so we could not test filter_by_type because that would require us to add factories for other models that can be actors. Future issues be handling this as well as the problem with get_by_actor() mentioned above.
